clients/cli/src/copy.rs

```diff
@@ -24,75 +25,237 @@ pub fn copy(path: PathBuf, import_dest: &str, edit: bool) {
+                CoreError::UiError(CreateFileAtPathError::FileAlreadyExists) => {
+                    println!("Input destination {} not available within lockbook, use --edit to overwrite the contents of this file!", import_dest)
                 }
                 CoreError::UiError(CreateFileAtPathError::NoAccount) => exit_with_no_account(),
                 CoreError::UiError(CreateFileAtPathError::NoRoot) => exit_with("No root folder, have you synced yet?", NO_ROOT),
-                CoreError::UiError(CreateFileAtPathError::DocumentTreatedAsFolder) => exit_with(&format!("A file along the target destination is a document that cannot be used as a folder: {}", import_dest_with_filename), DOCUMENT_TREATED_AS_FOLDER),
-                CoreError::UiError(CreateFileAtPathError::PathContainsEmptyFile) => exit_with(&format!("Input destination {} contains an empty file!", import_dest_with_filename), PATH_CONTAINS_EMPTY_FILE),
+                CoreError::UiError(CreateFileAtPathError::DocumentTreatedAsFolder) => println!("A file along the target destination is a document that cannot be used as a folder: {}", import_dest),
+                CoreError::UiError(CreateFileAtPathError::PathContainsEmptyFile) => println!("Input destination {} contains an empty file!", import_dest),
                 CoreError::UiError(CreateFileAtPathError::PathDoesntStartWithRoot) => exit_with("Import destination doesn't start with your root folder.", PATH_NO_ROOT),
                 CoreError::Unexpected(msg) => exit_with(&msg, UNEXPECTED_ERROR),
-            },
-        };
+            }
+        }
+    }
+}
+
+fn copy_file(path: &PathBuf, import_dest: &str, edit: bool, is_folder_copy: bool) {
+    let content_to_import = fs::read_to_string(&path);
+    let absolute_path_maybe = fs::canonicalize(&path);
+
+    match (content_to_import, absolute_path_maybe) {
+        (Ok(content), Ok(absolute_path)) => {
+            let import_dest_with_filename = if import_dest.ends_with('/') {
+                match absolute_path.file_name() {
+                    Some(name) => match name.to_os_string().into_string() {
+                        Ok(string) => format!("{}{}", &import_dest, string),
+                        Err(err) => exit_with(
+                            format!(
+                                "Unexpected error while trying to convert an OsString -> Rust String: {:?}",
+                                err
+                            )
+                                .as_str(),
+                            UNEXPECTED_ERROR,
+                        ),
+                    },
+                    None => exit_with(
+                        "Import target does not contain a file name!",
+                        UNEXPECTED_ERROR,
+                    ),
+                }
+            } else {
+                import_dest.to_string()
+            };
+
+            let file_metadata = match create_file_at_path(&get_config(), &import_dest_with_filename)
```

Comment:
Instead of calling `get_config` each time you need `Config`, this can be stored in a variable at the beginning of the function.